version: "3"

services:
  gitea:
    image: docker.gitea.com/gitea:1.24.0
    environment:
      - USER_UID=1000
      - USER_GID=1000
      - GITEA__database__DB_TYPE=postgres
      - GITEA__database__HOST=db:5432
      - GITEA__database__NAME=gitea
      - GITEA__database__USER=james
      - GITEA__database__PASSWD=james123
    restart: always
    networks:
      - homelab
    volumes:
      - ./gitea:/data
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    ports:
      - "222:22"

  runner:
    image: docker.io/gitea/act_runner:nightly
    environment:
      CONFIG_FILE: /config.yaml
      GITEA_INSTANCE_URL: "http://gitea:3000"
      GITEA_RUNNER_REGISTRATION_TOKEN: "EsK59mlgCRWIguKXOQVILkzCCketrbb7B4bq8Jiy"
      GITEA_RUNNER_NAME: "gitea-runner-1"
    restart: always
    volumes:
      - ./config.yaml:/config.yaml
      - ./runner:/data
      - /var/run/docker.sock:/var/run/docker.sock
    networks:
      - homelab
    depends_on:
      - gitea

networks:
  homelab:
    name: homelab
    external: true